tests/ModelContextProtocol.Tests/Server/McpServerTests.cs

```diff
@@ -685,4 +685,50 @@ await transport.SendMessageAsync(new JsonRpcNotification
         await server.DisposeAsync();
         await serverTask;
     }
+
+    [Fact]
+    public async Task NotifyCancel_Should_Be_Handled()
+    {
+        // Arrange
+        TaskCompletionSource<JsonRpcNotification> notificationReceived = new();
+        TaskCompletionSource notificationIntercepted = new();
+        await using TestServerTransport transport = new();
+        transport.OnMessageSent = (message) =>
+        {
+            if (message is JsonRpcNotification notification && notification.Method == NotificationMethods.CancelledNotification)
+                notificationReceived.TrySetResult(notification);
+        };
+        var options = CreateOptions(new()
+        {
+            NotificationHandlers = [new(NotificationMethods.CancelledNotification, notification =>
+            {
+                InvalidOperationException exception = new("The sender of a notification shouldn't handle the notification.");
+                notificationIntercepted.TrySetException(exception);
+                return notificationIntercepted.Task;
+            })],
+        });
+        await using var server = McpServerFactory.Create(transport, options, LoggerFactory);
+
+        // Act
+        var token = TestContext.Current.CancellationToken;
+        Task serverTask = server.RunAsync(token);
+        await server.NotifyCancelAsync(
```

Comment:
Because you created the server around the TestServerTransport, aren't you just sending this cancellation notification back to itself?